### ISSUE TITLE
Get a stream of datetimes, starting from a datetime, with interval <time unit>

### ISSUE THREAD

**Issue by martinsvalin** [magplus/good_times#20]
Functions to generate a lazy list of datetimes, starting from an input datetime (default `now`), stepping forward of backward one <time_unit> per item.

Example:
```
iex> each_day_from({{2015, 2, 27}, {18, 30, 45}}) |> Enum.take 3
[{{2015, 2, 27}, {18, 30, 45}},
 {{2015, 2, 28}, {18, 30, 45}},
 {{2015, 3, 1}, {18, 30, 45}}]
```

```
iex> each_day_before({{2015, 2, 27}, {18, 30, 45}}) |> Enum.take 3
[{{2015, 2, 26}, {18, 30, 45}},
 {{2015, 2, 25}, {18, 30, 45}},
 {{2015, 2, 24}, {18, 30, 45}}]
```

**Comment by DevL** [comment]
Module name: `GoodTimes.TimeStreams`?

**Comment by martinsvalin** [comment]
If we're talking about breaking up this library into multiple modules, we should be looking at the functions we have already. 

`GoodTimes.TimeStreams` sounds a little like it refers to its implementation as a Stream, and it repeats "Time". But naming is hard. I'm going off to bed.

**Comment by DevL** [comment]
So we shove these type of functions into the generically named module `GoodTimes.Iterators`. :-D

**Comment by martinsvalin** [comment]
Closed by #23
